qiskit/transpiler/passes/layout/layout_2q_distance.py

```diff
@@ -52,12 +52,18 @@ def run(self, dag):
         if layout is None:
             return
 
+        if self.coupling_map is None or len(self.coupling_map.graph) == 0:
+            self.property_set[self.property_name] = 0
+            return
+
+        self.coupling_map.compute_distance_matrix()
+
         sum_distance = 0
 
         for gate in dag.two_qubit_ops():
-            physical_q0 = layout[gate.qargs[0]]
-            physical_q1 = layout[gate.qargs[1]]
+            physical_q0 = layout.v2p[gate.qargs[0]]
+            physical_q1 = layout.v2p[gate.qargs[1]]
 
-            sum_distance += self.coupling_map.distance(physical_q0, physical_q1) - 1
+            sum_distance += self.coupling_map.dist_matrix[physical_q0][physical_q1] - 1
```

Comment:
This access can be by tuple, I think? `self.coupling_map.dist_matrix[physical_q0, physical_q1]` should avoid construction of a (more expensive) temporary object.

```python
In [21]: arr = np.arange(100).reshape(10, 10)

In [22]: %timeit arr[5][5]
177 ns ± 0.684 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [23]: %timeit arr[5, 5]
93.1 ns ± 0.639 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/9268df5513b4965ecf2ea457c8bf86ba697586f1